core/src/main/java/io/grpc/internal/PickFirstLeafLoadBalancer.java

```diff
@@ -180,10 +180,18 @@ public Status acceptResolvedAddresses(ResolvedAddresses resolvedAddresses) {
 
   @Override
   public void handleNameResolutionError(Status error) {
+    if (rawConnectivityState == SHUTDOWN) {
+      return;
+    }
+
     for (SubchannelData subchannelData : subchannels.values()) {
       subchannelData.getSubchannel().shutdown();
     }
     subchannels.clear();
+    if (addressIndex != null) {
+      addressIndex.updateGroups(null);
+    }
+    rawConnectivityState = IDLE;
```

Comment:
Am I right in believing this line is the main fix?

Can things work if we use TF here instead? That might be less prone to error.

Reply:
Yes, lines 191-193 are the critical part and I think having an appropriate rawConnectivityState state may be helpful and is definitely a good idea.  Changed it to be TF instead of IDLE.